File: standalone/dzen2-hlwm/python/pipehandler.01-init.py
import os
import subprocess
import time

import output

# ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----
# pipe

def content_init(monitor, pipe_dzen2_out):
    output.set_tag_value(monitor)
    output.set_windowtitle('')
        
    text = output.get_statusbar_text(monitor)
    pipe_dzen2_out.stdin.write(text + '\n')
    pipe_dzen2_out.stdin.flush()

def run_dzen2(monitor, parameters):  
    command_out  = 'dzen2 ' + parameters + ' -p'

    pipe_dzen2_out = subprocess.Popen(
            [command_out], 
            stdin  = subprocess.PIPE,
            shell  = True,
            universal_newlines=True
        )

    content_init(monitor, pipe_dzen2_out)
    pipe_dzen2_out.stdin.close()

def detach_dzen2_debug(monitor, parameters):
    run_lemon(monitor, parameters)

def detach_dzen2(monitor, parameters):
    # in case of debugging purpose, 
    # uncomment all the fork related lines.
    pid_dzen2 = os.fork()
    
    if pid_dzen2 == 0:
        try:
            run_dzen2(monitor, parameters)
            os._exit(1)
        finally:
            import signal
            os.kill(pid_dzen2, signal.SIGTERM)

def detach_transset():
    pid_transset = os.fork()
    
    if pid_transset == 0:
        try:
            time.sleep(1)
            os.system('transset .8 -n dzentop >/dev/null')            
            os._exit(1)
        finally:
            import signal
            os.kill(pid_transset, signal.SIGTERM)

File: standalone/dzen2-hlwm/ruby/pipehandler.01-init.rb
require_relative 'output'

# ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----
# pipe

def content_init(monitor, dzen2_stdin)
  set_tag_value(monitor)
  set_windowtitle('')
      
  text = get_statusbar_text(monitor)
  dzen2_stdin.puts(text)
end

def run_dzen2(monitor, parameters)
  command_out  = 'dzen2 ' + parameters + ' -p'
  IO.popen(command_out, "w") do |io_dzen2| 
    content_init(monitor, io_dzen2)        
    io_dzen2.close()    
  end
end

def detach_dzen2(monitor, parameters)
  # warning: Signal.trap is application wide
  Signal.trap("PIPE", "EXIT")
    
  pid = fork { run_dzen2(monitor, parameters) }
  Process.detach(pid)
end

def detach_transset()
  pid = fork do
    sleep(1)
    system('transset .8 -n dzentop >/dev/null')        
  end
    
  Process.detach(pid)
end
